server/controllers/DataAccessLayer.js

```diff
@@ -226,11 +227,12 @@ exports.ReadReportsFile = function() {
         // For each chat message
         for (let i = 0; i < chatMessages.length; i++) {
             // Filter and grab the username and message
-            let username = chatMessages[i].split(/,(.+)/)[0]
+            let userID = chatMessages[i].split(/,(.+)/)[0];
+            let user = cachedUsers.find(x => x.id === userID);
             let message = chatMessages[i].split(/,(.+)/)[1];
 
-            // Create a ChatMessage object
-            messages.push(new ChatMessage(username, message));
+            // Create a ChatMessage object with an empty tableID
+            messages.push(new ChatMessage("", user, message));
```

Comment:
just to be clear, reports doesn't care about the table id?

Reply:
Nope, the tableID is only relevant when storing chatMessages in the cache, and submitting reports. Once a report is submitted, the tableID isn't needed

Reply:
gotcha, everything else looks good to me